docs/building_blocks_for_flexible_content.md

```diff
@@ -0,0 +1,571 @@
+    blocks:
+    - type: heading
+      content: Rorem ipsum dolor sit amet, consectetur adipiscing elit.
+      path: http://gov.uk
+    - type: statistics
+      x_axis_label: "X Axis"
+      y_axis_label: "Y Axis"
+      csv_file: "data_one.csv"
+      data_source_link: https://www.ons.gov.uk/economy/grossdomesticproductgdp/timeseries/ihyq/qna
+```
+
+#### Featured
+
+A featured block renders a vertical column of blocks on the left of a large (but not full-width) image. It uses a picture element to allow multiple image versions for different devices.
+
+##### Example
+
+```yaml
+- type: featured
+  image:
+    alt: example alt text
+    sources:
+      desktop: "landing_page/placeholder/desktop.png"
+      desktop_2x: "landing_page/placeholder/desktop_2x.png"
+      mobile: "landing_page/placeholder/mobile.png"
+      mobile_2x: "landing_page/placeholder/mobile_2x.png"
+      tablet: "landing_page/placeholder/tablet.png"
+      tablet_2x: "landing_page/placeholder/tablet_2x.png"
+  featured_content:
+    blocks:
+      - type: heading
+        content: Our tasks
+        path: /landing-page/tasks
+      - type: govspeak
+        content: |
+          <p>Lorem ipsum dolor sit amet. In voluptas dolorum vel veniam nisi et voluptate dolores id voluptatem distinctio. Et quia accusantium At ducimus quis aut voluptates iusto aut esse suscipit.</p>
+```
+
+#### Hero
+
+A hero block renders a vertical column of blocks in a two-thirds-width block on the lower-left of a full-width image. Like the featured block, it uses a picture element to allow multiple image versions for different devices.
+
+##### Example
+
+```yaml
+- type: hero
+  image:
+    alt: "Placeholder alt text"
+    sources:
+      desktop: "landing_page/placeholder/desktop.png"
+      desktop_2x: "landing_page/placeholder/desktop_2x.png"
+      mobile: "landing_page/placeholder/mobile.png"
+      mobile_2x: "landing_page/placeholder/mobile_2x.png"
+      tablet: "landing_page/placeholder/tablet.png"
+      tablet_2x: "landing_page/placeholder/tablet_2x.png"
+  hero_content:
+    blocks:
+      - type: heading
+        content: Rorem ipsum dolor sit
+      - type: govspeak
+        content: |
+          <p>Yorem ipsum dolor sit amet, consectetur
+          adipiscing elit. Nunc vulputate libero et velit
+          interdum, ac aliquet odio mattis class.</p>
+      - type: action_link
+        text: "Learn more about our goals"
+        href: "/landing-page/goals"
+```
+
+### Layout blocks
+
+Layout blocks are similar to compound blocks in that they contain nested blocks. They are used to arrange blocks on the page, and to make it easier to apply styling to the blocks.
+
+- [Blocks container](#blocks-container)
+- [Columns layout](#columns-layout)
+- [Grid Container](#grid-container)
+- [Two column layout](#two-column-layout)
+
+#### Blocks container
+
+A blocks container is used as an empty unstyled parent container to
+hold other elements. It is used when we don't want to want to create
```

Comment:
Why/when would we not want to create a row/grid layout?

Reply:
I pilfered that explanation from the [commit message](https://github.com/alphagov/frontend/commit/f15d1e35ed7140b4eac00497ccca7ffd88fbd735). Does the comment in the PR make sense? If so I can replicate it wholesale.

Reply:
Yes, that helps. I think the key is knowing that it's for situations where you've already defined your grid and columns and you're nesting things within that space.

Reply:
Oh that's a good explanation. I'll add it.